python/cudf/cudf/core/index.py

```diff
@@ -347,6 +347,56 @@ def dropna(self, how="any"):
         """
         return super().dropna(how=how)
 
+    def _clean_nulls_from_index(self):
+        """
+        Convert all na values(if any) in Index object
+        to `null` as a preprocessing step to `__repr__` methods.
+
+        This will involve changing type of Index object
+        to StringIndex but it is the responsibility of the `__repr__`
+        methods using this method to replace or handle representation
+        of the actual types correctly.
+        """
+        if self._values.has_nulls:
+            return cudf.Index(
+                self._values.astype("str").fillna("null"), name=self.name
+            )
+        else:
+            return self.copy()
```

Comment:
We don't require a deep copy and since this is as an internal method we can also return the same object itself instead of a shallow copy too - does that sound okay to you?